kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/credential/ExecCommandCredentialLoader.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.credential;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.kubernetes.client.openapi.config.KubernetesClientConfiguration;
+import io.micronaut.kubernetes.client.openapi.config.model.ExecConfig;
+import io.micronaut.kubernetes.client.openapi.config.model.ExecEnvVar;
+import io.micronaut.kubernetes.client.openapi.config.KubeConfig;
+import io.micronaut.kubernetes.client.openapi.credential.model.ExecCredential;
+import io.micronaut.serde.ObjectMapper;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.InputStream;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The credential loader which uses the exec command from the kube config to get credentials.
+ */
+@Singleton
+@BootstrapContextCompatible
+class ExecCommandCredentialLoader implements KubernetesCredentialLoader {
+    private static final Logger LOG = LoggerFactory.getLogger(ExecCommandCredentialLoader.class);
+
+    private static final long BUFFER_IN_MINUTES = 1L;
+
+    private final KubeConfig kubeConfig;
+    private final ObjectMapper objectMapper;
+
+    private ExecCredential execCredential;
```

Comment:
```suggestion
    private volatile ExecCredential execCredential;
```

Reply:
it is read outside a synchronized block so it needs to be volatile.

Reply:
Done